### PR TITLE
Add E2E test 

### PR DESCRIPTION
**Pull request by tanyutu**
## Explain the feature/fix
Add the E2Etest in root directory, it contains apitest and uitest

- apitest is totally done, related to issue #206 

- uitest is done based on the mockup-ui. it could be working once all things have been wrapped up, related to issue #207 

## Does this PR introduce a breaking change
No

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Was this feature approved?

### CODE REVIEW

**Comment by kunyan** [comment]
@tanyutu 

Here are some suggestion

1. The all e2e test files could be store in `/e2e` folder. each component test files could be

* /e2e/
  * api
  * router
  * manager 

2. I think you won't need add another node project, you can just install all your dependency to the root
3. Please do not push any `*.tar` file into SCM repo. except it is something really need. if you want to upload a tar as test file, you could use `npm pack` and pack your test files to a compress file
4. Your `json` file looks not have a good format. You should clone the project and run `npm install` first. it will setup pre-commit hook, which will fix all format issue automatically 


@sayak-sarkar @Jared-Sprague ^^ Is that make sense ?

File: E2Etest/uitest/cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (kunyan) [inline E2Etest/uitest/cypress/fixtures/example.json:1]: You should remove this

**Comment by kunyan** [comment]
@tanyutu 

As we just talked.

1. You can add following code to make a test compressed file before test code running

```json
{
  "pretest": "npm pack ./spa-example",
  "test": "node ./index"
}.

2. The e2e could be have 2 ways

* cli - command line test
* manager - UI test

You could put all UI e2e to `manager`.

I don't know if we should provide `api` e2e test or `cli` e2e test. 
You already have api e2e test code, I think you can put them into `api` root


@sayak-sarkar @Jared-Sprague let me know if you have any idea for e2e test file structure

**Comment by tanyutu** [comment]
I feel it would more reasonable to create a monorepo for e2e. @kunyan thoughts on it?

**Comment by sayak-sarkar** [comment]
@kunyan @tanyutu I agree with @kunyan and am of the opinion that we should for now simply have an `/e2e` directory on the root of the current repo for containing all the tests. So I'm thinking something like:-
```
/e2e
 |- api
 |- manager
 |- router
```

File: E2Etest/uitest/cypress/fixtures/example.json (hunk 1 above)
Comment (sayak-sarkar) [reply E2Etest/uitest/cypress/fixtures/example.json:1]: This file isn't needed and should be removed. Once the directory structure is modified, the rest of the PR looks good to me.

Comment (tanyutu) [reply E2Etest/uitest/cypress/fixtures/example.json]: removed

**Comment by tanyutu** [comment]
The all e2e test files could be store in /e2e folder. each component test files could be
/e2e/
api
router
manager
### done

I think you won't need add another node project, you can just install all your dependency to the root
### keep the origin as before due to the test script path issue not fixed yet if install dependency of the root, can do the improvement in next patch.

Please do not push any *.tar file into SCM repo. except it is something really need. if you want to upload a tar as test file, you could use npm pack and pack your test files to a compress file
### removed the .tar file
Your json file looks not have a good format. You should clone the project and run npm install first. it will setup pre-commit hook, which will fix all format issue automatically
### fixed.

**Comment by sayak-sarkar** [comment]
Looks good to me! Thanks @tanyutu :-)

**Pull request merged**